package.json

```
@@ -2,131 +2,150 @@
+      "require": "./dist/cjs/ethers.js"
     },
     "./public": {
-      "types": "./dist/public.d.ts",
-      "module": "./dist/public.mts",
-      "default": "./dist/public.js"
+      "types": "./dist/types/public.d.ts",
+      "import": "./dist/esm/public.js",
+      "require": "./dist/cjs/public.js"
     },
     "./test": {
-      "types": "./dist/test.d.ts",
-      "module": "./dist/test.mts",
-      "default": "./dist/test.js"
+      "types": "./dist/types/test.d.ts",
+      "import": "./dist/esm/test.js",
+      "require": "./dist/cjs/test.js"
     },
     "./utils": {
-      "types": "./dist/utils/index.d.ts",
-      "module": "./dist/utils/index.mts",
-      "default": "./dist/utils/index.js"
+      "types": "./dist/types/utils/index.d.ts",
+      "import": "./dist/esm/utils/index.js",
+      "require": "./dist/cjs/utils/index.js"
     },
     "./wallet": {
-      "types": "./dist/wallet.d.ts",
-      "module": "./dist/wallet.mts",
-      "default": "./dist/wallet.js"
+      "types": "./dist/types/wallet.d.ts",
+      "import": "./dist/esm/wallet.js",
+      "require": "./dist/cjs/wallet.js"
     },
     "./window": {
-      "types": "./dist/window.d.ts",
-      "module": "./dist/window.mts",
-      "default": "./dist/window.js"
+      "types": "./dist/types/window.d.ts",
+      "import": "./dist/esm/window.js",
+      "require": "./dist/cjs/window.js"
     },
     "./package.json": "./package.json"
   },
-  "main": "dist/index.js",
-  "module": "dist/index.mjs",
-  "types": "dist/index.d.ts",
-  "sideEffects": false,
+  "typesVersions": {
+    "*": {
+      "abi": [
+        "./dist/types/abi.d.ts"
+      ],
+      "accounts": [
+        "./dist/types/accounts/index.d.ts"
+      ],
+      "chains": [
+        "./dist/types/chains.d.ts"
+      ],
+      "contract": [
+        "./dist/types/contract.d.ts"
+      ],
+      "ens": [
+        "./dist/types/ens.d.ts"
+      ],
+      "ethers": [
+        "./dist/types/ethers.d.ts"
+      ],
+      "public": [
+        "./dist/types/public.d.ts"
+      ],
+      "test": [
+        "./dist/types/test.d.ts"
+      ],
+      "utils": [
+        "./dist/types/utils/index.d.ts"
+      ],
+      "wallet": [
+        "./dist/types/wallet.d.ts"
+      ],
+      "window": [
+        "./dist/types/window.d.ts"
+      ]
+    }
+  },
+  "scripts": {
+    "anvil": "source .env && anvil --fork-url $VITE_ANVIL_FORK_URL --fork-block-number $VITE_ANVIL_BLOCK_NUMBER --block-time $VITE_ANVIL_BLOCK_TIME",
+    "bench": "vitest bench --no-threads",
+    "bench:ci": "CI=true vitest bench --no-threads",
+    "build": "pnpm run clean && pnpm run build:cjs && pnpm run build:esm+types && size-limit",
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --outDir ./dist/cjs && echo > ./dist/cjs/package.json '{\"type\":\"commonjs\"}'",
+    "build:esm+types": "tsc --project tsconfig.build.json --module es2015 --outDir ./dist/esm --declaration --declarationMap --declarationDir ./dist/types",
```

Should this be esnext?

```suggestion
    "build:esm+types": "tsc --project tsconfig.build.json --module esnext --outDir ./dist/esm --declaration --declarationMap --declarationDir ./dist/types",
```

---

There would be no difference in our case:https://www.typescriptlang.org/tsconfig#module

Hence why I chose the "lowest" boundary